crates/matrix-sdk-sled/src/state_store.rs

```diff
@@ -150,8 +174,129 @@ const TIMELINE_METADATA: &str = "timeline-metadata";
 #[cfg(feature = "experimental-timeline")]
 const TIMELINE: &str = "timeline";
 
+const ALL_DB_STORES: &[&str] = &[
+    ACCOUNT_DATA,
+    SYNC_TOKEN,
+    DISPLAY_NAME,
+    INVITED_USER_ID,
+    JOINED_USER_ID,
+    MEDIA,
+    MEMBER,
+    PRESENCE,
+    PROFILE,
+    ROOM_ACCOUNT_DATA,
+    #[cfg(feature = "experimental-timeline")]
+    ROOM_EVENT_ID_POSITION,
+    ROOM_EVENT_RECEIPT,
+    ROOM_INFO,
+    ROOM_STATE,
+    ROOM_USER_RECEIPT,
+    ROOM,
+    STRIPPED_INVITED_USER_ID,
+    STRIPPED_JOINED_USER_ID,
+    STRIPPED_ROOM_INFO,
+    STRIPPED_ROOM_MEMBER,
+    STRIPPED_ROOM_STATE,
+    #[cfg(feature = "experimental-timeline")]
+    ROOM_EVENT_ID_POSITION,
+    #[cfg(feature = "experimental-timeline")]
+    TIMELINE_METADATA,
+    #[cfg(feature = "experimental-timeline")]
+    TIMELINE,
+];
+const ALL_GLOBAL_KEYS: &[&str] = &[VERSION_KEY];
+
 type Result<A, E = SledStoreError> = std::result::Result<A, E>;
 
+#[derive(Builder, Debug, PartialEq)]
+#[builder(name = "SledStoreBuilder", build_fn(skip))]
+pub struct SledStoreBuilderConfig {
+    path: PathBuf,
+    passphrase: String,
+    #[builder(default = "MigrationConflictStrategy::BackupAndDrop")]
+    migration_conflict_strategy: MigrationConflictStrategy,
+}
+
+impl SledStoreBuilder {
+    pub fn build(&mut self) -> Result<SledStore> {
+        let is_temp = self.path.is_none();
+
+        let mut cfg = Config::new().temporary(is_temp);
+
+        let path = if let Some(path) = &self.path {
+            let path = path.join("matrix-sdk-store");
```

Comment:
```suggestion
            let path = path.join("matrix-sdk-state");
```

Although the name is inaccurate when it also contains the crypto store, it's the previous name used by `inner_open_with_path` in old L298. This would create a new database alongside the old one.

Another solution might be to rename the old database.